org.eclipse.xtext.tests/src/org/eclipse/xtext/util/MergeableManifest2Test.java

```diff
@@ -982,46 +989,84 @@ public void hachCode_differ_becauseOfEntry() throws Exception {
+			" org.antlr.runtime;bundle-version=\"[3.2.0,3.2.1)\"," + NL + 
+			" org.eclipse.xtend.lib;bundle-version=\"2.14.0\"," + NL + 
+			" org.eclipse.emf.common," + NL + 
+			" org.eclipse.xtext.common.types," + NL + 
+			" org.eclipse.xtext.xbase.lib;bundle-version=\"2.14.0\"" + NL + 
+			"Bundle-RequiredExecutionEnvironment: JavaSE-1.8" + NL + 
+			"Export-Package: org.eclipse.xtext.example.homeautomation.ruleEngine.util," + NL + 
+			" org.eclipse.xtext.example.homeautomation.ruleEngine," + NL + 
+			" org.eclipse.xtext.example.homeautomation.ruleEngine.impl," + NL + 
+			" org.eclipse.xtext.example.homeautomation," + NL + 
+			" org.eclipse.xtext.example.homeautomation.parser.antlr.lexer," + NL + 
+			" org.eclipse.xtext.example.homeautomation.parser.antlr.internal," + NL + 
+			" org.eclipse.xtext.example.homeautomation.validation," + NL + 
+			" org.eclipse.xtext.example.homeautomation.jvmmodel," + NL + 
+			" org.eclipse.xtext.example.homeautomation.scoping," + NL + 
+			" org.eclipse.xtext.example.homeautomation.serializer," + NL + 
+			" org.eclipse.xtext.example.homeautomation.parser.antlr," + NL + 
+			" org.eclipse.xtext.example.homeautomation.services," + NL + 
+			" org.eclipse.xtext.example.homeautomation.formatting2" + NL + 
+			"Import-Package: org.apache.log4j" + NL + 
+			"Automatic-Module-Name: org.eclipse.xtext.example.homeautomation" + NL;
 		// @formatter:on
 		MergeableManifest2 manifest = newManifest(content);
 		assertEquals(content, write(manifest));
 	}
 
+	/**
+	 * In this final test we read all "META-INF/MANIFEST.MF" files from
+	 * classpath. We the:
+	 * <ol>
+	 * <li>Read each file once as "manifest0"</li>
+	 * <li>Write it as string once and read the result again as "manifest1"</li>
+	 * <li>Compare "manifest0" and "manifest1"</li>
+	 * <li>Execute some sanity checks to ensure we do not compare empty files or
+	 * similar</li>
+	 * </ol>
+	 * This way we ensure we can read and write all manifest files.
+	 */
+	@Test
+	public void readWrite_allManifestsFromClasspath() throws Exception {
+		for (String content : readAllManifestsFromClasspath()) {
+			MergeableManifest2 manifest0 = newManifest(content);
+			MergeableManifest2 manifest1 = newManifest(write(manifest0));
+			assertEquals(manifest0, manifest1);
+			// start of sanity checks
+			assertTrue(content.length() > 0);
+			assertNotNull(manifest0.getMainAttributes().get(MergeableManifest2.MANIFEST_VERSION));
+		}
+	}
+
+	private List<String> readAllManifestsFromClasspath() throws IOException {
+		List<String> result = new ArrayList<>();
+		Enumeration<URL> resources = MergeableManifest2Test.class.getClassLoader().getResources("META-INF/MANIFEST.MF");
+		while (resources.hasMoreElements()) {
+			try (InputStreamReader reader = new InputStreamReader(resources.nextElement().openStream())) {
+				String content = CharStreams.toString(reader);
+				result.add(content);
+			} catch (IOException e) {
+				// ignore manifest files that can not be read
```

Comment:
Can we whitelist the broken manifest files? Otherwise we will mask some cases. Eg its hard to judge if even more than one single case was treated successfully right now